scripts/wrappers/start.sh

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bash
+
+set -eux
+
+source "${OPS_ROOT}"/helpers/snap-logger.sh "daemon"
+
+usage() {
+cat << EOF
+usage: start.sh --init-security yes --admin-password ...
+To be ran / setup once per cluster.
+--init-security   (Optional)    Enum of either: yes (default), no . Should be ran ONCE per cluster IF security enabled.
+--admin-password  (Optional)    Passphrase of the admin key
+--help                          Shows help menu
+EOF
+}
+
+# TODO: change defaults and register the service arguments in an arguments file
+
+
+# Args
+init_security=""
+admin_password=""
+
+
+# Args handling
+function parse_args () {
+    local LONG_OPTS_LIST=(
+        "init-security"
+        "admin-password"
+        "help"
+    )
+    # shellcheck disable=SC2155
+    local opts=$(getopt \
+      --longoptions "$(printf "%s:," "${LONG_OPTS_LIST[@]}")" \
+      --name "$(readlink -f "${BASH_SOURCE}")" \
+      --options "" \
+      -- "$@"
+    )
+    eval set -- "${opts}"
+
+    while [ $# -gt 0 ]; do
+        case $1 in
+            --init-security) shift
+                init_security=$1
+                ;;
+            --admin-password) shift
+                admin_password=$1
+                ;;
+            --help) usage
+                exit
+                ;;
+        esac
+        shift
+    done
+}
+
+function set_defaults () {
+    if [ -z "${init_security}" ] || [ "${init_security}" != "no" ]; then
+        init_security="yes"
+    fi
+}
+
+
+parse_args "$@"
+set_defaults
+
+
+# system config
+if ! snapctl is-connected systemd-write;
+then
+    echo "Please run the following command: sudo snap connect opensearch:systemd-write"
+    echo "Then run: sudo snap restart opensearch.daemon"
+    exit 1
+fi
+
+
+# start
+"${SNAP}"/usr/bin/setpriv \
+    --clear-groups \
+    --reuid snap_daemon \
+    --regid snap_daemon -- \
+    "${OPENSEARCH_HOME}"/bin/opensearch
+
+
+# run security_admin tool if security enabled
+# How to set passphrases
+# TODO only run once per cluster
```

Comment:
I'm guessing you're looking to write a check that bails with an error if this has already been run? Do we want to include a `reset` functionality too?

Reply:
I am not sure about raising an error here. I think that in an automated context like juju, this is fairly straightforward - as we only need to make sure to add the `init-security=yes` setting from the leader unit on install and this would be our safeguard. 
However, since this snap might be used elsewhere, we should indicate in our README that this must only be ran once per cluster and any subsequent run will override the security settings, but ultimately users might want for various reasons to force override the security indexes of opensearch, in which case they probably know best why so they should still be able to.
What do you think?